crypto/cmp/cmp_msg.c

```diff
@@ -20,6 +20,46 @@
 #include <openssl/err.h>
 #include <openssl/x509.h>
 
+OSSL_CMP_MSG *OSSL_CMP_MSG_new(OSSL_LIB_CTX *libctx, const char *propq)
+{
+    OSSL_CMP_MSG *msg = NULL;
+
+    msg = (OSSL_CMP_MSG *)ASN1_item_new_ex(ASN1_ITEM_rptr(OSSL_CMP_MSG),
+                                           libctx, propq);
+    if (!ossl_cmp_msg_set0_libctx(msg, libctx, propq)) {
```

Comment:
Is the ossl_cmp_msg_set0_libctx() needed here?  ASN1_item_new_ex() getes passed the libctx,propq?

Reply:
It is still needed. ASN1_item_new_ex() only knows about ASN1 things and passes the libctx/propq to any callbacks etc that it  needs to call during construction. But it can't populate any things in the structure not directly ASN1 related. These must be filled in later.